src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrVersionFileMerger.cs

```diff
@@ -0,0 +1,318 @@
+        string vmrPreviousRef,
+        string vmrCurrentRef,
+        string mapping,
+        string jsonRelativePath,
+        bool allowMissingFiles = false);
+
+    Task<VersionFileChanges<DependencyUpdate>> MergeVersionDetails(
+        Codeflow lastFlow,
+        Codeflow currentFlow,
+        string mappingName,
+        ILocalGitRepo targetRepo,
+        string targetBranch);
+}
+
+public class VmrVersionFileMerger : IVmrVersionFileMerger
+{
+    private readonly IGitRepoFactory _gitRepoFactory;
+    private readonly ILogger<VmrVersionFileMerger> _logger;
+    private readonly IVmrInfo _vmrInfo;
+    private readonly ILocalGitRepoFactory _localGitRepoFactory;
+    private readonly IVersionDetailsParser _versionDetailsParser;
+    private readonly IDependencyFileManager _dependencyFileManager;
+    private readonly IVersionPropertySelectorRegistry _mergerRegistry;
+    private const string EmptyJsonString = "{}";
+
+    public VmrVersionFileMerger(
+        IGitRepoFactory gitRepoFactory,
+        ILogger<VmrVersionFileMerger> logger,
+        IVmrInfo vmrInfo,
+        ILocalGitRepoFactory localGitRepoFactory,
+        IVersionDetailsParser versionDetailsParser,
+        IDependencyFileManager dependencyFileManager,
+        IVersionPropertySelectorRegistry mergerRegistry)
+    {
+        _gitRepoFactory = gitRepoFactory;
+        _logger = logger;
+        _vmrInfo = vmrInfo;
+        _localGitRepoFactory = localGitRepoFactory;
+        _versionDetailsParser = versionDetailsParser;
+        _dependencyFileManager = dependencyFileManager;
+        _mergerRegistry = mergerRegistry;
+    }
+
+    public async Task MergeJsonAsync(
+        Codeflow lastFlow,
+        ILocalGitRepo targetRepo,
+        string targetRepoPreviousRef,
+        string targetRepoCurrentRef,
+        ILocalGitRepo vmr,
+        string vmrPreviousRef,
+        string vmrCurrentRef,
+        string mappingName,
+        string jsonRelativePath,
+        bool allowMissingFiles = false)
+    {
+        var targetRepoPreviousJson = await GetJsonFromGit(targetRepo, jsonRelativePath, targetRepoPreviousRef, allowMissingFiles);
+        var targetRepoCurrentJson = await GetJsonFromGit(targetRepo, jsonRelativePath, targetRepoCurrentRef, allowMissingFiles);
+
+        var vmrJsonPath = VmrInfo.GetRelativeRepoSourcesPath(mappingName) / jsonRelativePath;
+        var vmrPreviousJson = lastFlow is Backflow
+            ? await GetJsonFromGit(vmr, vmrJsonPath, vmrPreviousRef, allowMissingFiles)
+            : targetRepoPreviousJson;
+        var vmrCurrentJson = await GetJsonFromGit(vmr, vmrJsonPath, vmrCurrentRef, allowMissingFiles);
+
+        var targetRepoChanges = SimpleConfigJson.Parse(targetRepoPreviousJson).GetDiff(SimpleConfigJson.Parse(targetRepoCurrentJson));
+        var vmrChanges = SimpleConfigJson.Parse(vmrPreviousJson).GetDiff(SimpleConfigJson.Parse(vmrCurrentJson));
+
+        var mergedChanges = MergeJsonChanges(targetRepoChanges, vmrChanges);
```

Comment:
We can pass the comparer function here as an argument
```suggestion
        var mergedChanges = MergeJsonChanges(targetRepoChanges, vmrChanges, JsonVersionPropertySelector.Select);
```